components/message_identifier_router.jsx

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+import {joinChannel} from 'mattermost-redux/actions/channels';
+import {getUser, getUserByUsername, getUserByEmail} from 'mattermost-redux/actions/users';
+
+import ChannelView from 'components/channel_view';
+import UserStore from 'stores/user_store.jsx';
+import ChannelStore from 'stores/channel_store.jsx';
+import TeamStore from 'stores/team_store.jsx';
```

Comment:
This is minor and up to you since the code was already doing this, but it would nice if we could use redux in here instead of these flux stores

Reply:
> Ah, to make router container using redux would be better.

Yeah, actually it was same means above.

Reply:
@jwilander Can you check the `channel_identifier_router.jsx` on the latest commit?

Still use the flux stores for selection utilities so I'm not sure it makes sense for your comment. (ex: getByName)
And `doChannelChange()` seems to not work properly.

Reply:
Nah, that's fine. We can deal with that in another PR in the future